lua/entities/gmod_wire_expression2/core/custom/prop.lua

```
@@ -454,23 +478,233 @@ end
+	if inertia:IsZero() then return end
+	this:SetInertia(inertia)
+end
+
+e2function void bone:setBuoyancy(number buoyancy)
+	if not boneVerify(self, this) then return end
+	this:SetBuoyancyRatio( math.Clamp(buoyancy, 0, 1) )
+end
+
+e2function void bone:setPhysicalMaterial(string material)
+	if not boneVerify(self, this) then return end
+	if not ValidAction(self, this, "physmat") then return end
+	this:SetMaterial(material)
+end
+
+e2function void bone:setVelocity(vector velocity)
+	local ent, index = boneVerify(self, this)
+	if not ValidAction(self, ent, "velocitynxt", this, index) then return end
+	this:SetVelocity(velocity)
+	ent:PhysWake()
+end
+
+e2function void bone:setVelocityInstant(vector velocity)
+	local ent, index = boneVerify(self, this)
+	if not ValidAction(self, ent, "velocityins", this, index) then return end
+	this:SetVelocityInstantaneous(velocity)
+	ent:PhysWake()
+end
+
+e2function void bone:setAngVelocity(vector velocity)
+	local ent, index = boneVerify(self, this)
+	if not ValidAction(self, ent, "angvelnxt", this, index) then return end
+	this:SetAngleVelocity(velocity)
+	ent:PhysWake()
+end
+
+e2function void bone:setAngVelocityInstant(vector velocity)
+	local ent, index = boneVerify(self, this)
+	if not ValidAction(self, this, "angvelinst", ent, index) then return end
+	this:SetAngleVelocityInstantaneous(velocity)
+	ent:PhysWake()
+end
+
+__e2setcost(500)
+
+-- This code was leveraged from Garry's Mod. Perhaps it would be a bit cleaner with a slight rewrite.
+e2function void entity:makeStatue(enable)
+	if not ValidAction(self, this, "statue") then return end
+	if this:GetNWBool("IsStatue") and enable ~= 0 then return
+	elseif not this:GetNWBool("IsStatue") and enable == 0 then return end
+	
+	
+	local timeout = this.statueTimeout
+	if timeout and timeout.time > CurTime() then
+		self:throw("Wait before repeated statue operations", false)
+		return
+	end
+
+	local bones = this:GetPhysicsObjectCount()
+	if bones < 2 then return end
```

Maybe should throw an error saying you can only use the function on ragdolls?